walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,99 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.List;
+import java.util.Stack;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Cases
+    // Options for no attendees - should be available all day.
+    if (request.getAttendees().size() == 0) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+    // Duration longer than a day will return empty array.
+    if (request.getDuration() > TimeRange.getTimeInMinutes(23, 59)) {
+      return Arrays.asList();
+    }
+    // Check if there are availabilities for all attendees
+    List<String> attendees = new ArrayList<>();
+    attendees.addAll(request.getAttendees());
+    attendees.addAll(request.getOptionalAttendees());
```

Comment:
I don't think optional attendees need to be considered here? You treat them the same as mandatory attendees.